src/battle_ai_switch_items.c

```diff
@@ -1785,14 +1785,65 @@ static bool32 CanAbilityTrapOpponent(u16 ability, u32 opponent)
         return FALSE;
 }
 
+static bool32 IsFreeSwitch(bool32 isSwitchAfterKO, u32 battlerSwitchingOut, u32 opposingBattler)
+{
+    u32 i;
+    bool32 movedSecond = GetBattlerTurnOrderNum(battlerSwitchingOut) > GetBattlerTurnOrderNum(opposingBattler) ? TRUE : FALSE;
```

Comment:
I wonder if using `GetBattlerTurnOrderNum` in AI calcs is safe. I know it can have values from a previous turn so that might mess up calcs